app/views/application/_header.html.erb

```diff
@@ -12,23 +11,48 @@
       <button class="usa-menu-btn"><%= t('shared.header.menu') %></button>
     </div>
     <nav aria-label="<%= t('shared.header.primary') %>" class="usa-nav">
-      <button class="usa-nav__close">
-        <%= image_tag "uswds/dist/img/usa-icons/close.svg", role: "img", alt: t('shared.header.close') %>
-      </button>
-      <ul class="usa-nav__primary usa-accordion">
-        <% if user_signed_in? %>
-          <li class="usa-nav__primary-item">
-            <%= link_to "Admin", "/admin/", class: "usa-link" %>
-          </li>
-          <li class="usa-nav__primary-item">
-            <%= button_to "Sign out", destroy_user_session_path, method: :delete %>
-          </li>
-        <% else %>
-          <li class="usa-nav__primary-item">
-            <%= button_to "Sign in with Github", user_github_omniauth_authorize_path %>
-          </li>
-        <% end %>
-      </ul>
+      <div class="usa-nav__inner">
+        <button class="usa-nav__close">
+          <%= image_tag "uswds/dist/img/usa-icons/close.svg", role: "img", alt: t('shared.header.close') %>
+        </button>
+        <ul class="usa-nav__primary usa-accordion">
+          <% @pages.each_with_index do |p, index| %>
+            <% if @user_signed_in || p.public? %>
```

Comment:
[suggestion] this should be `<% if user_signed_in? || p.public? %>`

In ruby, instance variables (`@name`) are `nil` if they haven't been set, which evaluates to false in an `if`